docs/changelog/112650.yaml

```diff
@@ -0,0 +1,6 @@
+pr: 112650
+summary: Handle ArithmeticException(long overflow) in dateFieldMapper
+area: Search
+type: bug
+issues:
+  - 112483
```

Comment:
I totally agree with your point. Now I think, the only use case of this patch can be providing better error message to the user (given that we catch it in `DateFieldMapper` as well, and convert to `ElasticsearchParseException` to make it 4xx)

Reply:
IllegalArgumentException does normally get converted to a 400 error here: https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/ExceptionsHelper.java#L64 . Any reason why this one specifically would lead to a 500 instead?

Reply:
I wasn't aware of this and assumed IllegalArgumentException returns 500 based on previous comment. As that's not the case, I think the patch should work without additonal modification

Reply:
Yeah maybe I was confusing it with another exception.

Reply:
Do we have a test that verifies this? Don't just believe me :)

Reply:
I can't verify the full circle here because the original issue did not contain reproducing example and I am not sure how to reproduce it from scratch. But this suggests @javanna is correct: https://github.com/elastic/elasticsearch/blob/f412de7e8ec93631b224e7d350907852e4433121/server/src/main/java/org/elasticsearch/ExceptionsHelper.java#L63 so while I'd be more happy if we had full cycle test I think this is ok.